### ISSUE TITLE
Unable to install 

### ISSUE THREAD

**Issue by reActionHank** [jcrodriguez1989/autokeras#2]
tried installing it on 2 separate machines, but I get the same error:

Error in install_keras(method = method, conda = conda, version = keras, : You should call install_keras() only in a fresh R session that has not yet initialized Keras and TensorFlow (this is to avoid DLL in use errors during installation)

Tried a couple of different things, creating new anaconda environments, reinstalling python, installing miniconda, etc.

I can install the keras and tensorflow packages separately w/o much problems, just the autokeras is an issue.

If you have any idea what to do would appreaciate it.

**Comment by jcrodriguez1989** [comment]
Hi @reActionHank , thanks for your feedback.
I am trying to reproduce your issue, so far, I am guessing you are using Windows.
So, in my Windows 10 VM I am not being able to even install keras.
Have you tried starting an R session and doing
```r
remotes::install_github("rstudio/keras")
library("keras")
install_keras()
```
Is this working for you?

thanks

**Comment by reActionHank** [comment]
Hey, thanks for the fast response. I've managed to install keras, but it
was difficult. It worked with an older version of Miniconda3 (4.11).

This also helped: https://github.com/rstudio/keras/issues/256

Unfortunately, I'm not able to do the same with autokeras.

**Comment by jcrodriguez1989** [comment]
@reActionHank I am not being able to reproduce your issue since I am not a Windows user, and my VM needs a lot of libraries installation.
Your `Error in install_keras(method = method, ...` error is not an autokeras code error. If you can reproduce it by creating a new empty env and try `install_keras()` then you might post it as a keras issue, this would be helpful for both autokeras and keras R libraries :) .
However, as a workaround for autokeras, you could manually install the Python Auto-Keras library.
You could try activating your keras working environment (remember it must have Python 3.6) and do from prompt:
```bash
conda install pytorch torchvision -c pytorch
pip install autokeras
```

**Comment by jcrodriguez1989** [comment]
@reActionHank I think what you noticed, might be an issue. Could you please reinstall autokeras `remotes::install_github("jcrodriguez1989/autokeras")` and re try doing `install_autokeras()`?
thanks!

In `install_autokeras()` was using `reticulate::py_config()` to get current Python version, and using this function initializes Python, and thus when `keras::install_keras()` calls `reticulate::py_available()` it returns `TRUE` and then the error you noted

```r
> reticulate::py_available()
[1] FALSE
> reticulate::py_config()$version
[1] "3.6"
> reticulate::py_available()
[1] TRUE
```

**Comment by reActionHank** [comment]
Hey, this is the error I got after trying the new install:

> install_autokeras()

Remove all packages in environment
C:\Users\Desktop\MINICO~1\envs\r-tensorflow:


## Package Plan ##

  environment location: C:\Users\Desktop\MINICO~1\envs\r-tensorflow


The following packages will be REMOVED:

    _tflow_select:          2.3.0-mkl
    absl-py:                0.7.1-py37_0                  conda-forge
    asn1crypto:             0.24.0-py37_1003              conda-forge
    astor:                  0.7.1-py_0                    conda-forge
    blas:                   1.0-mkl
    ca-certificates:        2019.3.9-hecc5488_0           conda-forge
    certifi:                2019.3.9-py37_0               conda-forge
    cffi:                   1.12.3-py37hb32ad35_0         conda-forge
    chardet:                3.0.4-py37_1003               conda-forge
    cryptography:           2.6.1-py37hb32ad35_0          conda-forge
    freetype:               2.10.0-h5db478b_0             conda-forge
    gast:                   0.2.2-py_0                    conda-forge
    grpcio:                 1.16.1-py37h351948d_1
    h5py:                   2.9.0-nompi_py37h3cb27cb_1102 conda-forge
    hdf5:                   1.10.4-nompi_hcc15c50_1106    conda-forge
    icc_rt:                 2019.0.0-h0cc432a_1
    idna:                   2.8-py37_1000                 conda-forge
    intel-openmp:           2019.3-203
    jpeg:                   9c-hfa6e2cd_1001              conda-forge
    keras:                  2.2.4-0
    keras-applications:     1.0.7-py_0                    conda-forge
    keras-base:             2.2.4-py37_0
    keras-preprocessing:    1.0.9-py_0                    conda-forge
    libblas:                3.8.0-5_mkl                   conda-forge
    libcblas:               3.8.0-5_mkl                   conda-forge
    liblapack:              3.8.0-5_mkl                   conda-forge
    libmklml:               2019.0.3-0
    libpng:                 1.6.37-h7602738_0             conda-forge
    libprotobuf:            3.7.1-h1a1b453_0              conda-forge
    libtiff:                4.0.10-h36446d0_1001          conda-forge
    markdown:               2.6.11-py_0                   conda-forge
    mkl:                    2019.3-203
    numpy:                  1.16.2-py37h8078771_1         conda-forge
    olefile:                0.46-py_0                     conda-forge
    openssl:                1.1.1b-hfa6e2cd_2             conda-forge
    pillow:                 6.0.0-py37h9a613e6_0          conda-forge
    pip:                    19.0.3-py37_0
    protobuf:               3.7.1-py37he025d50_0          conda-forge
    pycparser:              2.19-py37_1                   conda-forge
    pyopenssl:              19.0.0-py37_0                 conda-forge
    pyreadline:             2.1-py37_1000                 conda-forge
    pysocks:                1.6.8-py37_1002               conda-forge
    python:                 3.7.3-h8c8aaf0_0
    pyyaml:                 5.1-py37hfa6e2cd_0            conda-forge
    requests:               2.21.0-py37_1000              conda-forge
    scipy:                  1.2.1-py37h29ff71c_0
    setuptools:             41.0.0-py37_0
    six:                    1.12.0-py37_1000              conda-forge
    sqlite:                 3.27.2-he774522_0
    tensorboard:            1.13.1-py37_0                 conda-forge
    tensorflow:             1.13.1-mkl_py37h9463c59_0
    tensorflow-base:        1.13.1-mkl_py37hcaf7020_0
    tensorflow-estimator:   1.13.0-py37h39e3cac_0         conda-forge
    tensorflow-hub:         0.4.0-py_0                    conda-forge
    tensorflow-probability: 0.5.0-py_1                    conda-forge
    termcolor:              1.1.0-py_2                    conda-forge
    tk:                     8.6.9-hfa6e2cd_1001           conda-forge
    urllib3:                1.24.2-py37_0                 conda-forge
    vc:                     14.1-h0510ff6_4
    vs2015_runtime:         14.15.26706-h3a45250_0
    werkzeug:               0.15.2-py_0                   conda-forge
    wheel:                  0.33.1-py37_0
    win_inet_pton:          1.1.0-py37_0                  conda-forge
    wincertstore:           0.2-py37_0
    yaml:                   0.1.7-hfa6e2cd_1001           conda-forge
    zlib:                   1.2.11-h2fa13f4_1004          conda-forge

Creating r-tensorflow conda environment for TensorFlow installation...
Solving environment: ...working... done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.6.14

Please update conda by running

    $ conda update -n base -c defaults conda


sqlite-3.28.0        | 945 KB    | ########## | 100%
certifi-2019.3.9     | 155 KB    | ########## | 100%
## Package Plan ##

  environment location: C:\Users\Desktop\MINICO~1\envs\r-tensorflow

  added / updated specs:
    - python=3.7


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    sqlite-3.28.0              |       he774522_0         945 KB
    certifi-2019.3.9           |           py37_0         155 KB
    ------------------------------------------------------------
                                           Total:         1.1 MB

The following NEW packages will be INSTALLED:

    ca-certificates: 2019.1.23-0
    certifi:         2019.3.9-py37_0
    openssl:         1.1.1b-he774522_1
    pip:             19.0.3-py37_0
    python:          3.7.3-h8c8aaf0_0
    setuptools:      41.0.0-py37_0
    sqlite:          3.28.0-he774522_0
    vc:              14.1-h0510ff6_4
    vs2015_runtime:  14.15.26706-h3a45250_0
    wheel:           0.33.1-py37_0
    wincertstore:    0.2-py37_0


Downloading and Extracting Packages
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done

#
# To activate this environment, use:
# > activate r-tensorflow
#
# To deactivate an active environment, use:
# > deactivate
#
# * for power-users using bash, you must source
#

Solving environment: ...working... failed

PackagesNotFoundError: The following packages are not available from
current channels:

  - autokeras==0.3.7

Current channels:

  - https://conda.anaconda.org/conda-forge/win-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/free/win-64
  - https://repo.anaconda.com/pkgs/free/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/pro/win-64
  - https://repo.anaconda.com/pkgs/pro/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.


Error: Error 1 occurred installing packages into conda environment
r-tensorflow

**Comment by jcrodriguez1989** [comment]
thanks!
So, the bug is over. Now it is working at it should, and it is saying that the conda env is going to use Python 3.7, which is not supported by autokeras. And there it appears the new issue you are having.
I have been checking this, and Python 3.7 is being used because tensorflow R library uses this Python version if using default tensorflow version (1.3).
However, if you specify tensorflow 1.2. It will have other problems related to pytorch and etc.
To summarise, I would recommend you installing [Python autokeras](http://autokeras.com/) manually in your conda env.

**Comment by loheden** [comment]
@jcrodriguez1989 , how do you install autokeras manually from source? I am not able to install it on MAC OS M1.
